tiledb/api/c_api/object/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+#
+# tiledb/api/c_api/object/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(common NO_POLICY_SCOPE)
+
+list(APPEND SOURCES
+  object_api.cc
+)
+gather_sources(${SOURCES})
+
+#
+# Object library for other units to depend upon
+#
+add_library(capi_object OBJECT ${SOURCES})
+target_link_libraries(capi_object PUBLIC export)
+
+#
+# Test-compile of object library ensures link-completeness
+#
+add_executable(compile_capi_object EXCLUDE_FROM_ALL)
```

Comment:
This probably doesn't link because of a `StorageManager` dependency.

Reply:
It didn't. I moved the extra object APIs back to tiledb.{h,cc} for the time being. Once we clear up the StorageManager dependency issues they'll be easy enough to move back to the new API layout.